.github/workflows/maven.yml

```
@@ -0,0 +1,74 @@
+name: Check Maven
+run-name: Run checks for PR ${{ github.event.pull_request.title }}
+on:
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - 'isyfact-standards-doc'
+env:
+  MAVEN_CLI_OPTS: '--batch-mode --errors --fail-at-end -T 1C'
+
+jobs:
+  Compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: maven
+      - name: Compile
+        run: mvn compile $MAVEN_CLI_OPTS -Dcheckstyle.skip
+
+  Test:
+    runs-on: ubuntu-latest
+    needs: [Compile]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: maven
+      - name: Test
+        run: mvn test $MAVEN_CLI_OPTS -Dcheckstyle.skip
+      - name: Create Summary
+        run: |
+          echo "| Module | Total Coverage |" >> $GITHUB_STEP_SUMMARY
+          echo "| --- | --- |" >> $GITHUB_STEP_SUMMARY
+          COUNTER=0
+          SUM=0
+          for REPORT in $(find . -path \*target/site/jacoco/index.html); do
+            MODULE=$(grep -o '<title>[^%].*</title>'  "$REPORT"| sed 's/<\/\?title>//g')
+            COVERAGE=$(grep -o 'Total[^%]*%' "$REPORT" | sed 's/<.*>/ /; s/Total//; s/ //; s/%//')
+            echo "| $MODULE | $COVERAGE % |" >> $GITHUB_STEP_SUMMARY
+            COUNTER=$((COUNTER+1))
+            SUM=$((SUM+COVERAGE))
+          done
+          [[ $COUNTER -gt 0 ]] && TOTAL_COVERAGE=$((SUM/COUNTER)) || TOTAL_COVERAGE=0
+          echo "| Overall | $TOTAL_COVERAGE % |" >> $GITHUB_STEP_SUMMARY
+      - name: Upload Report
+        uses: actions/upload-artifact@v3
+        with:
+          name: Jacoco Test Report
+          path: ${{ github.workspace }}/**/target/site/jacoco/*
+
+  CodeQuality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: maven
+      - name: Checkstyle
+        run: mvn checkstyle:checkstyle-aggregate $MAVEN_CLI_OPTS
```

Optionen sollten immer vor den goals / phases stehen:
mvn [options] [<goal(s)>] [<phase(s)>]
```suggestion
        run: mvn $MAVEN_CLI_OPTS checkstyle:checkstyle-aggregate
```

---

Wurde angepasst.